pygnmi/client.py

```diff
@@ -673,20 +680,35 @@ def set(
                 )
 
             if gnmi_extension:
-                gnmi_message_request = SetRequest(
-                    prefix=protobuf_prefix,
-                    delete=del_protobuf_paths,
-                    update=update_msg,
-                    replace=replace_msg,
-                    extension=[gnmi_extension],
-                )
+                if prefix is None and target is None:
+                    gnmi_message_request = SetRequest(
+                        delete=del_protobuf_paths,
+                        update=update_msg,
+                        replace=replace_msg,
+                        extension=[gnmi_extension],
+                    )
+                else:
+                    gnmi_message_request = SetRequest(
+                        prefix=protobuf_prefix,
+                        delete=del_protobuf_paths,
+                        update=update_msg,
+                        replace=replace_msg,
+                        extension=[gnmi_extension],
+                    )
             else:
-                gnmi_message_request = SetRequest(
-                    prefix=protobuf_prefix,
-                    delete=del_protobuf_paths,
-                    update=update_msg,
-                    replace=replace_msg,
-                )
+                if prefix is None and target is None:
+                    gnmi_message_request = SetRequest(
+                        delete=del_protobuf_paths,
+                        update=update_msg,
+                        replace=replace_msg,
+                    )
+                else:
+                    gnmi_message_request = SetRequest(
+                        prefix=protobuf_prefix,
+                        delete=del_protobuf_paths,
+                        update=update_msg,
+                        replace=replace_msg,
+                    )
```

Comment:
It will help to have the flexibility to send an empty SET request. Better than not having it. 
Also, as stated in [Section 3.4](https://github.com/openconfig/reference/blob/master/rpc/gnmi/gnmi-specification.md#34-modifying-state), 
```A SetRequest specifying an empty set of paths MUST NOT be treated as an error by the target.```
Previously, the smallest request that could be generated was:
```
prefix {
}
```
This is not an empty request.